Feat: Add styling to projects in second deadline

Cank256 opened the pull request:
Add styling to a project that has reached a second deadline.
See screen shot below (I did not have a project in the second deadline but changed the class of the project available to show the styling implementation)

<img width="1206" alt="Second Dealine styling" src="https://github.com/vinsky001/ALX-DarkMode-project/assets/15633901/bbc284dc-93df-4604-a653-82cdcf5f2b5a">

OxOneBeing commented:
Good one Caleb! I'd like to know if there are distinct classes for these deadlines?

Cank256 commented:
Yes, the project that goes into the second deadline has the **in_second_deadline** class.

MartyOfMCA reviewed (approved):
This would be of great help to our users. Well done Caleb!

MartyOfMCA reviewed (changes requested):
Kindly visit this link as well... There's projects pending for the second deadline as well

[Projects page](https://intranet.alxswe.com/projects/current)

Cank256 commented:
I have added the styling for this as well. I had to add a JS script that looks for the text "deadline for a second chance before" within the different project DIVs and then appends the in_second_deadline to them so that they can change styling when the page loads.

See screen shot below:

<img width="949" alt="Second Dealine Projects View" src="https://github.com/vinsky001/ALX-DarkMode-project/assets/15633901/2fd04c42-1f05-40fd-a22f-9880e19f81c9">

MartyOfMCA commented on css/alx-dark.css:
```diff
@@ -872,10 +878,24 @@ body.dark .tab-pane .well {
+  background-color: var(--col-second-deadline) !important;
+}
+
+body.dark .panel .list-group-item.in_second_deadline a {
+  color: var(--col-second-deadline-link) !important;
```
I'd suggest you add some more changes to make `.list-group-item.in_second_deadline a` stand out as a hyperlink. You can also add some properties to change the hyperlink when the user hovers on the link just as all the other hyperlinks do

Have a look at the attached photos showcasing some styles for the various states of the hyperlinks

When you is not hovering on the link:
![image](https://github.com/vinsky001/ALX-DarkMode-project/assets/104218489/e3bb291d-65e8-42d7-820f-dcb04bf60bb4)

When user is hovering on the link:
![image](https://github.com/vinsky001/ALX-DarkMode-project/assets/104218489/6244ccfe-6d0a-4d87-ad6f-6283c48ec97c)

MartyOfMCA commented on js/content-script.js:
```diff
@@ -69,3 +69,29 @@ const toggleLight = function () {
     }
   });
 })();
+
+document.addEventListener('DOMContentLoaded', function() {
+  var pendingLogsDiv = document.querySelector('.pending_logs');
+  
+  // Check if the div with class "pending_logs" exists
+  if (pendingLogsDiv) {
+    var listGroupItems = pendingLogsDiv.querySelectorAll('.list-group-item');
+    
+    // Loop through each list-group-item element
+    for (var i = 0; i < listGroupItems.length; i++) {
+      var listGroupItem = listGroupItems[i];
+      var bpiStatusSpan = listGroupItem.querySelector('.bpi-status');
+      
+      // Check if list-group-item has a child element with class "bpi-status"
+      if (bpiStatusSpan) {
+        var strongElement = bpiStatusSpan.querySelector('strong');
+        
+        // Check if bpi-status span has a child <strong> element with the word "second deadline"
+        if (strongElement && strongElement.innerText.includes('deadline for a second chance before')) {
+          // Add the class "in_second_deadline" to the respective list-group-item element
+          listGroupItem.classList.add('in_second_deadline');
+        }
+      }
+    }
+  }
+});
```
Great work here!

Cank256 commented:
I have added the on hover styling for the anchor tags and changed the script for appending the in_second_deadline class to look at list-group-item as the parent element instead of the pending_logs div element.

MartyOfMCA commented on css/alx-dark.css:
```diff
@@ -884,16 +884,24 @@ body.dark .panel .list-group-item.in_second_deadline {
+
+body.dark .panel .list-group-item.in_second_deadline a:hover {
+  color: var(--col-light-400) !important;
+}
```
I think this rule should go to the pending logs styles section on line 893. Just so we can spend less time searching for rules for components

MartyOfMCA commented on css/alx-dark.css:
```diff
@@ -884,16 +884,24 @@ body.dark .panel .list-group-item.in_second_deadline {
+}
 /* Dashboard styles ends here */
 
 /* Pending logs styles starts here */
```
And also change the description from Pending logs to something like second deadline... we'd easily understand that

Cank256 replied:
I have implemented this also

MartyOfMCA reviewed (approved):
Thanks Caleb
